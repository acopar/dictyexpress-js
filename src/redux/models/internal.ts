import {
    DataDifferentialExpression,
    GOEnrichmentJson,
    GOEnrichmentNode,
    Relation,
    Storage,
} from '@genialis/resolwe/dist/api/types/rest';
import { Feature } from '@genialis/resolwe/dist/api/types/modules';
import { VariantType, SnackbarKey } from 'notistack';
import { Renderers } from 'vega';
import { LayoutBreakpoint } from 'components/genexpress/common/constants';

export type Gene = Pick<
    Feature,
    'feature_id' | 'aliases' | 'name' | 'full_name' | 'description' | 'source' | 'species'
>;

export type RelationsById = {
    [_: number]: Relation;
};

export type GenesById = {
    [_: string]: Gene;
};

export type BasketInfo = {
    id: string;
    source: string;
    species?: string;
    type: 'gene';
};

export type GeneSet = {
    dateTime: Date;
    genesNames: string[];
};

export type GeneExpression = {
    [geneId: string]: number;
};

export type SamplesExpressionsById = {
    [sampleId: number]: GeneExpression;
};

export type VolcanoPoint = {
    geneId: string; // Feature (gene) id
    logFcValue: number;
    logProbValue: number;
    logProbFiniteValue: number;
    probValue: number;
};

export type Thresholds = {
    pValue: number;
    pValueLog: number;
    fc: number;
    fcLog: number;
};

export type DifferentialExpression = {
    logfc_threshold: number;
    prob_field: string;
    prob_threshold: number;
    up_regulated: number;
    down_regulated: number;
} & DataDifferentialExpression &
    Partial<Storage['json']>;

export type DifferentialExpressionsById = {
    [differentialExpressionId: number]: DifferentialExpression;
};

export type AspectValue = 'BP' | 'CC' | 'MF';
export type Aspect = {
    value: AspectValue;
    label: string;
};

export type EnhancedGOEnrichmentJson = {
    tree: {
        [aspectSlug: string]: GOEnrichmentRow[];
    };
} & Omit<GOEnrichmentJson, 'tree'>;

export type GOEnrichmentRow = {
    path: string[];
    children?: GOEnrichmentRow[];
    score_percentage: number;
    gene_associations: string[];
} & Omit<GOEnrichmentNode, 'children'>;

// If key is undefined, it will be generated in the reducer.
export type SnackbarNotificationContent = {
    key?: SnackbarKey;
    message: string;
    variant?: VariantType;
    action?: (key: number) => void;
};

// TODO, is this the correct way of going about making only one property required?
export type SnackbarNotification = {
    key: SnackbarKey;
} & Pick<SnackbarNotificationContent, 'message' | 'variant' | 'action'>;

export type SnackbarNotifications = SnackbarNotification[];

export type DictyRenderers = Exclude<Renderers, 'none'>;

export type BreakpointsCols = { [key in keyof typeof LayoutBreakpoint]: number };
