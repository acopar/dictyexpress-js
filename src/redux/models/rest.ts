export type BasketAddSamplesResponse = {
    id: string;
    modified: string;
    ignored: number[];
    permitted_organisms: string[];
    permitted_sources: string[];
    conflict_organisms: string[];
    conflict_sources: string[];
};
