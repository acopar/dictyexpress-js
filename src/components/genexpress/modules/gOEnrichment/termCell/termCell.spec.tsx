import React from 'react';
import { customRender } from 'tests/test-utils';
import { generateGOEnrichmentRow } from 'tests/mock';
import { GOEnrichmentRow } from 'redux/models/internal';
import GOEnrichmentTermCell from './termCell';

describe('gOEnrichmentTermCell', () => {
    let goEnrichmentRow: GOEnrichmentRow;
    let asFragment: () => DocumentFragment;

    beforeEach(() => {
        goEnrichmentRow = generateGOEnrichmentRow(1);
        goEnrichmentRow.term_name = 'Test term';
        goEnrichmentRow.depth = 3;

        ({ asFragment } = customRender(<GOEnrichmentTermCell data={goEnrichmentRow} />));
    });

    it('should render as default snapshot', () => {
        expect(asFragment()).toMatchSnapshot();
    });
});
