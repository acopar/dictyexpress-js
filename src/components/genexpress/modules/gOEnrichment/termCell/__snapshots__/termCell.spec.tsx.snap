// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gOEnrichmentTermCell should render as default snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bqWxrE humzUX"
  >
    <span
      class="sc-ksBlkl eyAxZE"
    />
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root-ffjXHZ fQQvfO MuiSvgIcon-root MuiSvgIcon-fontSizeMedium sc-hBxehG HUEyD"
      data-testid="ArrowDropDownIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="m7 10 5 5 5-5z"
      />
    </svg>
    <span>
      Test term
    </span>
  </div>
</DocumentFragment>
`;
